File: bubble_sort.c
#include <stdio.h>
#include <stdbool.h>

void swap(int *xp, int *yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void printArray(int arr[], int size)
{
	for(int i = 0; i < size; i++)
		printf("%d ", arr[i]);
	printf("\n");
}

void bubbleSort(int arr[], int n)
{
		int i, j;
		bool swapped;
		for(i = 0; i < n - 1; i++){
			swapped = false;
			for(j = 0; j < n - i - 1; j++){
				if(arr[j] > arr[j + 1]){
					swap(&arr[j], &arr[j + 1]);
					swapped = true;					
				}
			}
			if(swapped == false){
				break;
			}
		}
}

int main()
{
	int arr[] = {45, 6, 3, -9, 0, 333};
	int n = sizeof(arr) / sizeof(arr[0]);
	printf("Unsorted array: \n");
	printArray(arr, n);
	bubbleSort(arr, n);
	printf("Sorted array: \n");
	printArray(arr, n);
	return 0;
}

File: sorting_methods_js/bubble_sort.js
let bubbleSort = (inputArray)=> {
	let len = inputArray.length;
	let swapped;
	for(let i = 0; i < len; i++){
		swapped = false;
		for(let j = 0; j < len - i - 1; j++){
			if(inputArray[j] > inputArray[j + 1]){
				let tmp = inputArray[j];
				inputArray[j] = inputArray[j + 1];
				inputArray[j + 1] = tmp;
				swapped = true;
			}
		}
		if(swapped == false){
			break;
		}
	}
	return inputArray;
}
let array = [3, 5, 1, 4, 8, 2];
console.log(`Unsorted array: ${array}`);
console.log(`Sorted array: ${bubbleSort(array)}`);